File: src/main/java/fr/spoutnik87/musicbot_rest/repository/MediaRepository.java
package fr.spoutnik87.musicbot_rest.repository;

import fr.spoutnik87.musicbot_rest.model.Category;
import fr.spoutnik87.musicbot_rest.model.Media;
import fr.spoutnik87.musicbot_rest.model.MediaType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface MediaRepository extends JpaRepository<Media, Long> {

  List<Media> findByMediaType(MediaType mediaType);

  List<Media> findByCategory(Category category);

  Optional<Media> findByUuid(String uuid);
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/repository/MediaRepository.kt
package fr.spoutnik87.musicbot_rest.repository

import fr.spoutnik87.musicbot_rest.model.Category
import fr.spoutnik87.musicbot_rest.model.Media
import fr.spoutnik87.musicbot_rest.model.MediaType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface MediaRepository : JpaRepository<Media, Long> {
    fun findByUuid(uuid: String): Media?
    fun findByMediaType(mediaType: MediaType): List<Media>
    fun findByCategory(category: Category): List<Media>
}